code/modules/projectiles/updated_projectiles/guns/specialist.dm

```diff
@@ -38,7 +45,154 @@
 	Q.Attach(src)
 	update_attachables()
 	S.icon_state = initial(S.icon_state)
+	LT = image("icon" = 'icons/obj/items/projectiles.dmi',"icon_state" = "sniper_laser", "layer" =-LASER_LAYER)
+
+/obj/item/weapon/gun/rifle/sniper/M42A/Fire(atom/target, mob/living/user, params, reflex = 0, dual_wield)
+	if(!able_to_fire(user))
+		return
+	if(targetlaser_on)
+		if(!iscarbon(target))
+			return
+		if(laser_target)
+			laser_target.remove_laser()
+		to_chat(user, "<span class='danger'>You focus your targeting laser on [target]!</span>")
+		laser_target = target
+		targetlaser_on = FALSE
+		laser_target.apply_laser()
+		processing_objects.Remove(src) //So we don't accumulate additional processing.
+		processing_objects.Add(src)
+		return
+	return ..()
+
+
+/mob/living/carbon/proc/apply_laser()
+	return FALSE
+
+/mob/living/carbon/human/apply_laser()
```

Comment:
You can probably handle this more generically for mobs but no biggie if thats a lot of work

Reply:
TBH, I think we should use a standardized index of overlays between all mobs, but that is grossly out of scope.